tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,214 @@
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run open-vm-tools testing against VMware ESXi
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_autotest::esxi_utils;
+use Time::Local;
+
+my $VM_POWERED_ON  = 'Powered on';
+my $VM_POWERED_OFF = 'Powered off';
+
+sub run {
+    assert_script_run('rpmquery open-vm-tools');
+    systemctl('status vmtoolsd');
+    systemctl('status vgauthd');
+
+    my $vm_name = console('svirt')->name;
+    my $vm_id   = esxi_vm_get_vmid($vm_name);
+    my $vm_ip   = esxi_vm_public_ip($vm_id);
+
+    record_info('Guest Name', $vm_name);
+    record_info('Guest ID',   $vm_id);
+    record_info('Guest IP',   $vm_ip);
+
+    # Power Manangement Tests
+    do_power_management_tests($vm_id, $vm_ip);
+    # Networking Tests
+    do_networking_tests($vm_id, $vm_ip);
+    # Clock Sync Tests
+    do_clock_sync_tests($vm_name, $vm_id, $vm_ip);
+}
+
+sub check_power_state {
+    my ($vm_id, $vm_ip, $power_action, $if_pingable, $if_vm_state, $expect_vm_state) = @_;
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/$if_vm_state/) {
+        esxi_vm_power_cycle($vm_id, $power_action);
+        wait_for_vm_bootup($vm_ip, $if_pingable);
+    } else {
+        die "Not the correct VM state for guest $power_action.";
+    }
+    die "Failed to $power_action the guest." if (!esxi_vm_power_getstate($vm_id) =~ m/$expect_vm_state/);
+}
+
+sub wait_for_vm_bootup {
+    my $vm_ip       = shift;         # vm ip address
+    my $if_pingable = shift;         # 1 or 0
+    my $times       = shift // 6;    # the number of ping execution
+    my $count       = shift // 3;    # ping waits for the count
+    my $interval    = shift // 5;    # wait secs between each ping
+    my $cmd;
+
+    if ($if_pingable) {
+        $cmd = "ping -c$count -i$interval $vm_ip";
+    } else {
+        $cmd = "! ping -c$count -i$interval $vm_ip";
+    }
+
+    while ($times && console('svirt')->run_cmd($cmd)) {
+        $times--;
+    }
+}
+
+sub login_vm_console {
+    reset_consoles;
+    select_console('sut');
+    assert_screen('grub2', 90);
+    send_key 'ret';
+    select_console('root-console', timeout => 180);
+}
+
+sub set_guest_time {
+    my $n_hrs  = (sprintf "%02d", rand(24));
+    my $n_min  = (sprintf "%02d", rand(59));
+    my $n_sec  = (sprintf "%02d", rand(59));
+    my $g_time = script_output("date -u -s $n_hrs:$n_min:$n_sec +'\%F \%T'");
+
+    return $g_time;
+}
+
+sub get_timelc {
+    my ($time, $timelc, $sec, $min, $hour, $mday, $mon, $year);
+    $time = $_[0];
+
+    my @time = reverse(split /-|:|\s|\//, $time);
+    foreach (@time) {
+        s/^0//;
+        $timelc .= $_ . ",";
+    }
+    $timelc =~ s/,$//;
+    ($sec, $min, $hour, $mday, $mon, $year) = split /,/, $timelc;
+    $year -= 1900;
+    $mon  -= 1;
+    $timelc = timelocal($sec, $min, $hour, $mday, $mon, $year);
```

Comment:
What does $timelc stand for ? lc for local ? local clock ?

Reply:
It doesn't stand for a local time, the function timelocal() converts a formatted time to total seconds. Here it's just a variable name, no special meaning. It is only used to receive the value returned from the function timelocal().

Reply:
So 'lc" comes out of nothing ? If this is the case, why not use something more meaningful.

Reply:
OK, using  the variable name 'second_str' instead of 'timelc', and changing the function name to "get_second_str()".

Reply:
If this generates and returns epoch time, why not use get_time_epoch ?

Reply:
Great idea! Thanks for the suggestion. Will update with the name.